storage/local/persistence.go

```diff
@@ -159,7 +159,10 @@ func newPersistence(
 			return nil, err
 		}
 		if len(fis) > 0 && !(len(fis) == 1 && fis[0].Name() == "lost+found" && fis[0].IsDir()) {
-			return nil, fmt.Errorf("could not detect storage version on disk, assuming version 0, need version %d - please wipe storage or run a version of Prometheus compatible with storage version 0", Version)
+
+			unknownFiles := len(fis)
+
+			return nil, fmt.Errorf("could not detect storage version on disk; assuming version 0, need version %d; please wipe storage (removing %d unexpected entries) or run a version of Prometheus compatible with storage version 0", Version, unknownFiles)
```

Comment:
That's not quite right. Imagine you have a `lost+found` _and_ something else.

In different news, I don't think listing the number of entries is valuable information. Perhaps just add to the message that it could be triggered by files in the directory that have nothing to do with Prometheus.